lib/katello/engine.rb

```diff
@@ -132,9 +132,15 @@ class Engine < ::Rails::Engine
 
     config.to_prepare do
       Katello::CandlepinEventListener.client_factory = proc do
+        settings = SETTINGS[:katello][:candlepin_events].deep_merge(
```

Comment:
You don't really need to deep merge here
```suggestion
        settings = SETTINGS[:katello][:candlepin_events].merge(
```

Reply:
This will always override the defaults, even if they don't exist within `SETTINGS`. In other words, you can drop the defaults in `candlepin_events` and also not override them anymore even if you wanted to.

The alternative is to do create a hash with the `SETTINGS[:ssl_*]` values, call `compact` to filter out `nil` values but they're typically used together. If you use the key from defaults but the cert from the settings it's very unlikely to work.

Any thoughts on this?

Reply:
There are no defaults for these values that I can find.

Reply:
Yes - you nailed it. I'll add the comment.

I do like the move to rely on defaults and only configure overrides. I think that will help and keep things cleaner in our container deployment.

Reply:
I still think the code comment to use `.merge()` instead of `.deep_merge()` stands.

Also, I'd still like to know if this would work:
```ruby
settings = ActiveRecord::Base.connection_pool.with_connection do
  SETTINGS[:katello][:candlepin_events].merge(
    ssl_key_file: Setting[:ssl_priv_key],
    ssl_cert_file: Setting[:ssl_certificate],
    ssl_ca_file: Setting[:ssl_ca_file]
  )
end
```

Reply:
You are wanting me to test if Setting can be used and things still work?

Reply:
Tested and updated, no issues were found.